data/data/powervs/dns/dns.tf

```diff
@@ -0,0 +1,52 @@
+data "ibm_dns_domain" "domain" {
+  name = var.cluster_domain
+}
+
+resource "ibm_dns_record" "bootstrap" {
+  count              = 1
+  data               = var.bootstrap_ip
+  domain_id          = data.ibm_dns_domain.domain.id
+  host               = "bootstrap.${var.cluster_id}"
+  responsible_person = "root.${var.cluster_domain}."
+  ttl                = 900
+  type               = "a"
+  tags               = [var.cluster_id, "${var.cluster_id}-bootstrap"]
+}
+resource "ibm_dns_record" "master" {
+  count              = length(var.master_ips)
+  data               = var.master_ips[count.index]
+  domain_id          = data.ibm_dns_domain.domain.id
+  host               = "master-${count.index}.${var.cluster_id}"
+  responsible_person = "root.${var.cluster_domain}."
+  ttl                = 900
+  type               = "a"
+  tags               = [var.cluster_id, "${var.cluster_id}-master"]
+}
+
+resource "ibm_dns_record" "api" {
+  data               = "${var.load_balancer_hostname}."
+  domain_id          = data.ibm_dns_domain.domain.id
+  host               = "api.${var.cluster_id}"
+  responsible_person = "root.${var.cluster_domain}."
+  ttl                = 900
+  type               = "cname"
+  tags               = [var.cluster_id, "${var.cluster_id}-api"]
+}
+resource "ibm_dns_record" "api-int" {
+  data               = "${var.load_balancer_int_hostname}."
+  domain_id          = data.ibm_dns_domain.domain.id
+  host               = "api-int.${var.cluster_id}"
+  responsible_person = "root.${var.cluster_domain}."
+  ttl                = 900
+  type               = "cname"
+  tags               = [var.cluster_id, "${var.cluster_id}-api-int"]
+}
+resource "ibm_dns_record" "apps" {
+  data               = var.load_balancer_hostname
```

Comment:
missed this the first go-round. does this need the same formatting (the period at the end of the hostname) as the others? or is it different since it's a wildcard record?

Reply:
Excellent catch :-). The fully qualified host/domain name in `cname` record need to have the dot char. Ref: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/dns_record